stgit/commands/rebase.py

```diff
@@ -267,10 +269,16 @@ def __do_rebase_interactive(repository, previously_applied_patches, check_merged
                     instructions[index].apply,
                 )
                 patch_name = new_patch_name
-        if instructions[index].apply:
-            post_rebase(stack, {patch_name}, 'rebase', check_merged)
 
         instructions = __perform_squashes(instructions, index, stack)
+        index += 1
+
+    post_rebase(
+        stack,
+        [instruction.patch_name for instruction in instructions if instruction.apply],
+        'rebase',
+        check_merged,
+    )
```

Comment:
I was trying to think of why I didn't do this in the first place (`apply` all the patches once at the end instead of doing them individually during the loop). 

Then I remembered that until https://github.com/stacked-git/stgit/pull/127, there was no way to `apply` a patch which had been `squashed` into as part of the same instruction set... the squash patch has the potential to be renamed as part of the squash! However with the addition of https://github.com/stacked-git/stgit/pull/127 we can be sure that we still have a 'valid' instruction set at this point -- all instructions which might modify a patch name (ie `squash` and `edit`) now update the `instructions` set to use the new patch name.